.github/workflows/build_deploy_and_test.yml

```
@@ -107,19 +107,51 @@ jobs:
       - name: setup-environment-vars
         run: /build-tools-ci/scripts/set-environment
 
+      - name: create .npmrc
+        run: |
+          echo @yalesites-org:registry=https://npm.pkg.github.com >> .npmrc
+          echo //npm.pkg.github.com/:_authToken="$GITHUB_TOKEN" >> .npmrc
       - name: run static tests
         run: |
           ./.ci/test/static/run
 
   # If needed, frontend assets compilation or similar steps
   # could be added here.
 
+  build_frontend:
+    # container:
+    #   image: quay.io/pantheon-public/build-tools-ci:8.x-php7.4
+    #   options: --user root
+    runs-on: ubuntu-latest
+    needs: [configure_env_vars, static_tests]
+    if: github.event_name == 'pull_request'
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+        with:
+          ref: ${{ github.head_ref || github.ref_name }}
+          fetch-depth: 0
+      - name: create .npmrc
+        env:
+          GITHUB_TOKEN: ${{ secrets.GH_TOKEN }}
+        run: |
+          echo @yalesites-org:registry=https://npm.pkg.github.com >> .npmrc
+          echo //npm.pkg.github.com/:_authToken="$GITHUB_TOKEN" >> .npmrc
+      - name: build frontend components
+        env:
+          BRANCH: ${{ github.head_ref || github.ref_name }}
```

```suggestion
          BRANCH: ${{ github.head_ref || github.ref_name }}
          YALESITES_BUILD_TOKEN: ${{ secrets.YALESITES_BUILD_TOKEN }}
```

Does adding this here fix the authentication issue?

---

We may not need to do lines 134-139 at all, if so. (We can keep the existing `.npmrc` file, and just pass the token as an environment variable like we do locally)

---

After I rebased from develop, `.npmrc` exists in the repo now, so this suggestion worked. Thanks!